packages/affiliates/libs/contracts.js

```diff
@@ -27,11 +28,76 @@ function decodeAttribution(
   data,
   delimiter = "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff000000000000000000000000"
 ) {
+  assert(data, "requires data to decode");
   return data.split(delimiter)[1];
 }
 
+// Just wraps abi to pass through to contract
+// Lookup by erc20 address
+function Erc20({ abi = getAbi("ERC20"), web3 }) {
+  assert(abi, "requires abi for erc20");
+  const contract = new web3.eth.Contract(abi);
+  function decimals(address) {
+    assert(address, "requires address");
+    contract.options.address = address;
+    return contract.methods.decimals().call();
+  }
+  return {
+    decimals
+  };
+}
+
+// Wrapper for some basic emp functionality.
+// Currently we just need token and collateral info
+// Lookup by emp address
+function Emp({ abi = getAbi("ExpiringMultiParty"), web3 } = {}) {
```

Comment:
I'm assuming this class is used for many different EMPs at the same time? That's the reason for not parameterizing it with an address up-front?

Even so, this might be a little simpler to follow if emp instances were 1:1 with this class, just so you wouldn't have to pass address into each function. Not strongly opinionated (as I haven't seen the code that uses it), but it's something to consider.

Reply:
yes thats right, i guess it could be designed either way, but its just instantiated once and used to gather info on several emps in a loop